Class/practice2.2/main.cpp

```diff
@@ -0,0 +1,104 @@
+    vector<vector<vector<int>>> pixels(16, vector<vector<int>>(16, vector<int>(3))); // 16x16 image pixels
+    
+    // Read input file
+    for (int i = 0; i < 16; ++i) {
+        // I hope that it is better to read file line by line here to ensure that format matches.
+        // then you can split the line into array of substrings with std::ranges split and std::ranges::to<std::vector<std::string>>
+        // x,y,z can be also split with ranges and deliimiter comma
+        for (int j = 0; j < 16; ++j) {
+            // It seems that you should encapsulate part of the logic into Pixel struct.
+            // This struct should have isValid method to validate pixel values are in range 0-255.
+            // Also you will be able to overload << and >> operators for Pixel struct
+            // which will allow you to make more readable code and will hide the format details
+            // For example:
+            // auto& curPixel = pixels[i][j];
+            // inputFile >> curPixel;
+            // if (!curPixel.isValid()) {
+            //     cerr << "Error: Invalid pixel values in line " << i + 1 << ", column " << j + 1 << "!" << endl;
+            //     return;
+            // }
+            char comma;
+            if (!(inputFile >> pixels[i][j][0] >> comma >> pixels[i][j][1] >> comma >> pixels[i][j][2])) {
+                cerr << "Error: Invalid input format in line " << i + 1 << ", column " << j + 1 << "!" << endl;
+                return;
+            }
+            if (pixels[i][j][0] < 0 || pixels[i][j][0] > 255 ||
+                pixels[i][j][1] < 0 || pixels[i][j][1] > 255 ||
+                pixels[i][j][2] < 0 || pixels[i][j][2] > 255) {
+                cerr << "Error: Pixel values out of range in line " << i + 1 << ", column " << j + 1 << "!" << endl;
+                return;
+            }
+        }
+    }
+    
+    // Check and modify pixels according to favorite color
+    for (int i = 0; i < 16; ++i) {
+        for (int j = 0; j < 16; ++j) {
+            if (pixels[i][j] == favoriteColor) {
+                // when you will encapsulate part of the logic into Pixel struct,
+                // you will be able to simplify this code.
+                if (i == 0 || j == 0) {
+                    pixels[i][j] = favoriteColor;
+                } else {
+                    pixels[i][j][0] = pixels[i][0][0];
+                    pixels[i][j][1] = pixels[0][j][1];
+                    pixels[i][j][2] = pixels[0][j][2];
+                }
+            }
+        }
+    }
+    
+    // Write output to file
+    for (int i = 0; i < 16; ++i) {
+        // you can use std::views::join_with to get that string of pixels and with overloaded << operator pixel serialization will be simpler
+        for (int j = 0; j < 16; ++j) {
+            outputFile << pixels[i][j][0] << "," << pixels[i][j][1] << "," << pixels[i][j][2] << " ";
```

Comment:
operator<< can be overloaded or std::formatter for pixel class. In the last case you will be able to do
std::print(outputFile, "{}", pixels[i][j]